### PR TITLE
Feature/base price and quantity sale order

### PR DESCRIPTION
**Pull request by mileo**
Special thanks to @pedrobaeza by the https://github.com/odoo/odoo/pull/1470

### CODE REVIEW

File: l10n_br_sale/sale.py [hunk 1]
```diff
@@ -297,6 +306,12 @@ def _prepare_invoice(self, cr, uid, order, lines, context=None):
 class SaleOrderLine(orm.Model):
     _inherit = 'sale.order.line'
 
+    def _calc_line_base_price(self, cr, uid, line, context=None):
```
Comment (danimaribeiro) [inline l10n_br_sale/sale.py:309]: Is it necessary to redefine those two methods? They are equal to the base sale.py.

Comment (mileo) [inline l10n_br_sale/sale.py:309]: @fernandomr No, the super can be called

Comment (danimaribeiro) [inline l10n_br_sale/sale.py:309]: What about remove for now? just implement when we need

File: l10n_br_sale_product/sale.py [hunk 2]
```diff
@@ -251,6 +261,12 @@ def onchange_amount_costs(self, cr, uid, ids, amount_costs=False):
 class SaleOrderLine(orm.Model):
     _inherit = 'sale.order.line'
 
+    def _calc_line_base_price(self, cr, uid, line, context=None):
+        return line.price_unit * (1 - (line.discount or 0.0) / 100.0)
```
Comment (mileo) [inline l10n_br_sale_product/sale.py:265]: @danimaribeiro You are talking about this method? Or the method in odoo core?

Comment (danimaribeiro) [inline l10n_br_sale_product/sale.py:265]: I'm talking about this method, the code is equal to odoo core, so I don't think it needs to be redefined here.

Comment (mileo) [inline l10n_br_sale_product/sale.py:265]: Yes, I agree with you. However we will have to wait a few years for the Baeza's pr be approved ....

Comment (danimaribeiro) [inline l10n_br_sale_product/sale.py:265]: Hm, I see now. The PR still open.
So this is ok.

**Comment by danimaribeiro** [comment]
https://github.com/odoo/odoo/pull/1470 Merged.

**Comment by rvalyi** [comment]
:+1:

File: l10n_br_sale/sale.py [hunk 3]
```diff
@@ -310,15 +325,25 @@ def _amount_line(self, cr, uid, ids, field_name, arg, context=None):
...
+                line.order_id.partner_id,
                 fiscal_position=line.fiscal_position)
+
             cur = line.order_id.pricelist_id.currency_id
-            res[line.id]['price_subtotal'] = cur_obj.round(cr, uid, cur, taxes['total'])
-            res[line.id]['price_gross'] = line.price_unit * line.product_uom_qty
-            res[line.id]['discount_value'] = res[line.id]['price_gross']-(price * line.product_uom_qty)
+            res[line.id]['price_subtotal'] = cur_obj.round(
+                cr, uid, cur, taxes['total'])
+            res[line.id]['price_gross'] = price * qty
```
Comment (mileo) [inline l10n_br_sale/sale.py:344]: :-1: Price gross use normal price_unit

**Pull request closed without merging**